### ISSUE TITLE
Using new `string_flag` settings in external repositories

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#10152]
With the new `string_flag` type, it would be nice if we could include some of these in rules repos, but currently it doesn't seem like you can reference them if they're in a WORKSPACE repo in your project. For example if we add this:

```python
string_flag(
    name = "ios_simulator_id",
    build_setting_default = "",
    visibility = ["//visibility:public"],
)
```

To rules_apple, and then try to set it with:

```
bazel test //... --@build_bazel_rules_apple//apple:ios_simulator_id=foo
```

We get this error:

```
ERROR: Unrecognized option: --@build_bazel_rules_apple//apple:ios_simulator_id=foo
```

I also tried using `//external` here with no luck.

### What's the output of `bazel info release`?

release 1.1.0

Related: https://github.com/bazelbuild/rules_apple/pull/648

**Comment by gregestren** [comment]
Let's consolidate these bugs into #11128.

When that one is fixed, it if doesn't address this we can re-open this.

**Comment by juliexxia** [comment]
I believe this is fixed by #10039

Random repo I used was

$ git clone git clone https://github.com/bazelbuild/rules_go
$ bazel build :default_nogo --@io_bazel_rules_go//go/config:incompatible_package_conflict_is_error=True
